server/api-util/idToken.js

```diff
@@ -0,0 +1,82 @@
+const crypto = require('crypto');
+const { default: fromKeyLike } = require('jose/jwk/from_key_like');
+const { default: SignJWT } = require('jose/jwt/sign');
+
+const radix = 10;
+const PORT = parseInt(process.env.REACT_APP_DEV_API_SERVER_PORT, radix);
+const rootUrl = process.env.REACT_APP_CANONICAL_ROOT_URL;
+const useDevApiServer = process.env.NODE_ENV === 'development' && !!PORT;
+
+const issuerUrl = useDevApiServer ? `http://localhost:${PORT}/api` : `${rootUrl}/api`;
+
+const clientId = process.env.CUSTOM_OIDC_CLIENT_ID;
```

Comment:
What is this environment variable?

Reply:
I think the biggest change I'd request is to remove these implementation-specific environment variables away from this file. 
- if CUSTOM_OIDC_CLIENT_ID is needed in implementing a new OpenId client, define it before calling functions from this file (i.e. use currying if necessary `exports.createIdToken = clientId =>  user => {`)

Reply:
This is the client id you set to your identity provider in Console. In practice, users are free to set them whatever they wish as it doesn't need to be e.g. same as the LinkedIn client id (but it could be). The only requirement is that this clientId matches the one in Console.

Reply:
Thanks for the explanation. 👍 

I'd suggest that we try to avoid "clientId" variable name and use something more clear instead (maybe idpClientID or oidcClientId or something?)

I'd also push all the environment variables that are implementation-specific away from this utility file (as I mentioned in the previous comment).

```
// Usage: createIdToken(process.env.CUSTOM_OIDC_CLIENT_ID)
exports.createIdToken = idpClientId => user => {
// etc.
}
```